### PR TITLE
Fix: #1377. Reboot digest-algorithm values registry.

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

- [x] reboots digest algorithm values registry
- [x] add registration policy for new algorithms #1533
- [x] deprecate non-crypto algorithms (unix/sum, crc*, adler)

### CODE REVIEW

**Comment by ioggstream** [comment]
@LPardue preliminary review pls :)

File: draft-ietf-httpbis-digest-headers.md [hunk 1]
```diff
@@ -412,67 +411,110 @@ All digest-algorithm token values are case-insensitive
...
+     Use "standard" for standardized algorithms without known problems;
+     "experimental", "obsoleted", or some other appropriate value
+     - e.g. according to the type and status of the primary document
+     in which the algorithm is defined;
+     "deprecated" when the algorithm is insecure or otherwise undesirable;
+     "reserved" when Digest algorithm references a reserved token value
+ - Description: the description of the digest-algorithm and its encoding
+ - Reference: a set of pointers to the primary documents defining the digest-algorithm
+
+Values to be added to this registry require IETF Review (see Section 4.8 of {{!RFC8126}}).
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:431]: IANA list the current registration procedure for this table as "RFC Required or Specification Required". Why the change in registration policy? It seems more restrictive than current.

File: draft-ietf-httpbis-digest-headers.md [hunk 2]
```diff
@@ -412,67 +411,110 @@ All digest-algorithm token values are case-insensitive
...
+The associated encoding for new digest-algorithms MUST either
+be represented as a quoted string
+or MUST NOT include ";" or "," in the character sets used for the encoding.
+
+Deprecated digest algorithms MUST NOT be used.
 
+The registry is initialized with the tokens listed below.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:438]: I suspect the ordering of values below is meant to imply "best to worse". I think it's going to be simpler to just present them in lexical ascending order, as the IANA table does today.

File: draft-ietf-httpbis-digest-headers.md [hunk 3]
```diff
@@ -412,67 +411,110 @@ All digest-algorithm token values are case-insensitive
...
 
   {: vspace="0"}
-  sha-256
-  : * Description: The SHA-256 algorithm [RFC6234].  The output of
+  sha-512
+  : * Digest Algorithm: sha-512
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:442]: The repetition of the algorithm in each entry is a bit jarring. Maybe we should just present all algorithms in a table form.

File: draft-ietf-httpbis-digest-headers.md [hunk 4]
```diff
@@ -412,67 +411,110 @@ All digest-algorithm token values are case-insensitive
...
-- "sha", see [IACR-2020-014] and {{?NO-SHA1=RFC6194}}.
-
-See the references above for further information.
+Registrations MUST include the following fields:
+
+ - Digest algorithm: the token value.
+     The registry can be used to reserve token values
+ - Status: the status of the algorithm.
+     Use "standard" for standardized algorithms without known problems;
+     "experimental", "obsoleted", or some other appropriate value
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:423]: lets remove obsoleted as an example here?

File: draft-ietf-httpbis-digest-headers.md (hunk 3 above)
Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:442]: Tables are an editorial hell... I'd rather strip the `Digest Algorithm:` line ;)

File: draft-ietf-httpbis-digest-headers.md (hunk 2 above)
Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:438]: Stronger to weaker ;) I have no preference though.

File: draft-ietf-httpbis-digest-headers.md (hunk 1 above)
Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:431]: I understood it was a suggestion from @mnot, but if I'm wrong it's ok to change this line.

File: draft-ietf-httpbis-digest-headers.md (hunk 4 above)
Comment (ioggstream) [inline draft-ietf-httpbis-digest-headers.md:423]:
```suggestion
     "experimental" or some other appropriate value
```

File: draft-ietf-httpbis-digest-headers.md (hunk 3 above)
Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:442]: fair point. I can live with duplicating

File: draft-ietf-httpbis-digest-headers.md (hunk 1 above)
Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:431]: I've forgotten that suggestion, where was it made?

Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:431]: I inferred that from this comment on the  `multihash` discussion https://github.com/httpwg/http-extensions/issues/1532#issuecomment-856532947 

If that's a blocker, we can remove this line and move the discussion to #1533: I think the important part is to land this PR :)

Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:431]: Yeah let's punt registration policy changes to a new issue

Comment (ioggstream) [inline draft-ietf-httpbis-digest-headers.md:431]:
```suggestion
```

**Review by LPardue — APPROVED** [review]
